KHWE-14: Implement the simple router

lubovgribiniyk opened the pull request.

lubovgribiniyk committed:
KHWE-14: Implement simple router in Main component

lubovgribiniyk committed:
Merge branch 'master' of https://github.com/ihor4568/spotty-vanilla into KHWE-14-router

lubovgribiniyk committed:
KHWE-14: Add NotFound component and handle dynamic routes

lubovgribiniyk committed:
Merge branch 'master' of https://github.com/ihor4568/spotty-vanilla into KHWE-14-router

# Conflicts:
#	src/components/Main/Main.js

lubovgribiniyk committed:
KHWE-14: Add /albums route

ihor4568 commented on src/components/Main/Main.js:
```diff
@@ -30,15 +31,48 @@ export class MainComponent {
+
+    const urlParts = pathname.split("/");
+
+    if (urlParts[0] === "songs" && urlParts[1] && urlParts.length === 2) {
```
please contact Ivan - we have agreed on different path format for share url

ihor4568 commented on src/components/NotFound/NotFound.html:
```diff
@@ -0,0 +1 @@
+<h1 class="not-found mdc-typography--headline4">404: Page Not Found</h1>
```
please make text `Not Found`

ihor4568 replied:
also make its color gray. And display it at the center of the screen

lubovgribiniyk committed:
Merge branch 'master' of https://github.com/ihor4568/spotty-vanilla into KHWE-14-router

# Conflicts:
#	package-lock.json

yevheniiIvanise commented on src/components/Main/Main.js:
```diff
@@ -30,15 +31,48 @@ export class MainComponent {
 
   addEventListeners() {
     this.sidebarList.addEventListener("click", this.handleListClick.bind(this));
```
bind handleListClick function in the constructor once. you should not call bind function every time event triggers

lubovgribiniyk replied:
`bind` is called only when function `addEventListeners` is called and bind will return new function which will be set us as an event handler for `click` event.
In our case `addEventListeners` will be called only once. So `bind` will be called only once.

ihor4568 replied:
actually not - it binds one time when addEventListeners function is being called

yevheniiIvanise commented on src/components/Main/Main.js:
```diff
@@ -30,15 +31,48 @@ export class MainComponent {
 
   addEventListeners() {
     this.sidebarList.addEventListener("click", this.handleListClick.bind(this));
+    window.addEventListener("popstate", this.handleStatePath.bind(this));
```
same

lubovgribiniyk committed:
KHWE-14: Improve router

- Improve the sidebar click handler
- Change path format for share url
- Fix Not Found component styles

yevheniiIvanise commented on src/components/Main/Main.js:
```diff
@@ -30,15 +31,55 @@ export class MainComponent {
-    if (e.target.closest(".main__about-link")) {
-      this.about.mount();
-    } else if (e.target.closest(".main__albums-link")) {
+    const { target } = e;
```
you use it once in your block code, so i am not sure it is necessary to destruct the object. why not simply access the appropriate property as is through the dot notation?

ihor4568 replied:
not sure if it makes code any worse

ihor4568 commented:
could you please also center not found vertically?

![image](https://user-images.githubusercontent.com/43406825/47901114-bc8b5800-de87-11e8-9f04-139fa7facb1f.png)

yev-ivanisenko committed:
KHWE-14: Change height of main components

yev-ivanisenko committed:
Merge branch 'master' of github.com:ihor4568/spotty-vanilla into KHWE-14-router

# Conflicts:
#	src/components/Main/Main.html
#	src/components/Main/Main.js
#	src/index.scss

yev-ivanisenko committed:
KHWE-14: Change styles for NotFound component
